### ISSUE TITLE
Command line options on winvnc4.exe

### ISSUE THREAD

**Issue by Dmole** [TigerVNC/tigervnc#105]
Currently one must edit the registry, and setting the password is prohibitively complicated.

**Comment by CendioOssman** [comment]
The Windows server has a configuration GUI that you should use for updating the settings. Is that not working for you?

**Comment by Dmole** [comment]
I am aware of the GUI.
I requested a CLI (like those found on builds of tigervnc for other OSs).
The request is to make tigervnc more easily usable in scripts.

**Comment by CendioOssman** [comment]
Ah, OK. I'm marking this as a feature request then.

**Comment by darkpixel** [comment]
Yup--totally wanted to automating scripts.  I'm having trouble finding documentation on any CLI args on the Windows version.  I'm trying to test starting a VNC server for every user that logs in to an RDS server so the helpdesk can easily offer remote assistance to the users.  (The built-in Windows Remote Assistance is pretty terrible.)

**Comment by avih** [comment]
Was there any progress on this?

I'm trying to create a package to help family/friends with a vnc server on windows which is launched automatically with the required settings (ports/password/connect/etc) regardless if any other vnc server is installed/running on that system - except if they clash in ports of course, but I'll use a different enough port with my instance.

Tiger vnc on windows does not seem to support such invocation as far as I can tell, or at least I couldn't find a way to list the options with `--help` or some such.

Unfortunately it seems that other vnc servers also depend on registry, and even those which do support some "application mode" still use the registry for $things and cannot be fully configured via CLI and/or a custom config file which is set via CLI.

Can TigerVNC support such scenario? If not currently, are there such plans/status?

Thanks

**Comment by CendioOssman** [comment]
No progress, no. We don't really have anyone working on WinVNC. We try to not actively break it, but no one is working on improvements at this point. Help is always welcome.

**Comment by Dmole** [comment]
BTW modern web browsers have desktop sharing now (no mouse though).

I wonder if VNC will run from the new WSL.
It would require hours of down time and a reboot,
lol Microsoft should just replace Windows with Linux (Like they may replace IE with Chrome).

**Comment by avih** [comment]
@CendioOssman thanks, appreciate the info.

**Comment by jursed** [comment]
Bumped into this thread by coincidence, since I was only focusing on `vncconfig` so far.

Given the current state, it would help to mention in the README (or some other obvious place?) that:
- The Windows server is not being actively maintained
- The Windows build of `vncconfig` does not (seem to?) honor any passed [CLI parameters](https://tigervnc.org/doc/vncconfig.html) and always opens a GUI window no matter what
- Users must alter the Windows registry settings at ``HKLM\SOFTWARE\TigerVNC\WinVNC4`` themselves
- There is [no complete documentation](https://github.com/TigerVNC/tigervnc/issues/571#issuecomment-359435882) of all registry key-value pairs at the moment
- The VNC password is hashed by DES, if anyone cares :)
